DANS-KNAW/gmh-registration-service
# Refactoring

**Fixes** 

[DD-169](https://drivenbydata.atlassian.net/browse/DD-169) - Develop BRI-side code/service for push-based registration of NBNs to BRI

**How to test**

- check out this PR
- in local MYSQL instance, create database using the sql script located at `database/nbnresolver_updated.sql`
- in MYSQL execute following commands:
`INSERT INTO registrant (registrant_groupid) VALUES ('dans');`
`INSERT INTO credentials (registrant_id, org_prefix, username, password) VALUES (LAST_INSERT_ID(), 'UI:13', 'dansusername', 'cnSnMTIo' );`
- run `mvn clean package jetty:run`
- copy applications.properties file from `src/test/resources/application.properties` to `src/main/resources`
- fill in MYSQL credentials in `src/main/resources/application.properties`
- go to `http://localhost:8009/swagger-ui`
- get api token with credentials from db by using the `POST /token` endpoint
- enter the returned token in the `BearerAuth` field under the `Authorize` button at the top of the page
- start with a `POST /nbn` request
